File: xianguoji-server/src/main/java/com/xianguoji/server/module/review/entity/Review.java
package com.xianguoji.server.module.review.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName(value = "review", autoResultMap = true)
public class Review {

    @TableId(type = IdType.AUTO)
    private Long id;
    private Long orderId;
    private Long orderItemId;
    private Long productId;
    private Long userId;
    private Integer rating;
    private Integer freshnessRating;
    private Integer valueRating;
    private Integer packageRating;
    private String content;
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<String> images;
    private Integer isAnonymous;
    private Integer isHidden;
    private String merchantReply;
    private LocalDateTime repliedAt;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    @TableField(fill = FieldFill.INSERT)
    private Long createBy;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updateBy;
}

File: merchant-front/src/api/types/review.ts
/**
 * 后端 ReviewVO（com.xianguoji.server.module.review.vo.ReviewVO）
 * 商家端列表与公共评价列表共用同一个 VO
 */
export interface AdminReviewVO {
  id: number;
  productId: number;
  userName: string;
  userAvatar: string | null;
  rating: number;
  freshnessRating: number | null;
  valueRating: number | null;
  packageRating: number | null;
  content: string;
  images: string[];
  isAnonymous: 0 | 1;
  merchantReply: string | null;
  repliedAt: string | null;
  createdAt: string;
}
